Mathlib/RingTheory/Finiteness/Small.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir, María-Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, Maria-Inés de Frutos-Fernandez
+-/
+import Mathlib.LinearAlgebra.Finsupp.LinearCombination
+import Mathlib.RingTheory.FiniteType
+import Mathlib.LinearAlgebra.DFinsupp
+import Mathlib.Algebra.Algebra.Subalgebra.Basic
+import Mathlib.LinearAlgebra.Basis.Cardinality
+import Mathlib.LinearAlgebra.StdBasis
+import Mathlib.RingTheory.Finiteness.Basic
+import Mathlib.Data.DFinsupp.Small
+
+/-! # Smallness properties of modules and algebras -/
+
+universe u
+
+namespace Submodule
+
+variable {R M : Type*} [Semiring R] [AddCommMonoid M] [Module R M]
+
+theorem small_sup {P Q : Submodule R M} (_ : Small.{u} P) (_ : Small.{u} Q) :
+    Small.{u} (P ⊔ Q : Submodule R M) := by
+  rw [Submodule.sup_eq_range]
+  exact small_range _
+
+theorem FG.small [Small.{u} R] (P : Submodule R M) (hP : P.FG) : Small.{u} P := by
+  rw [fg_iff_exists_fin_generating_family] at hP
+  obtain ⟨n, s, rfl⟩ := hP
+  rw [← Fintype.range_linearCombination]
+  apply small_range
```

Comment:
Could you add the [Module.Finite](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Finiteness/Defs.html#Module.Finite) version of this?